### PR TITLE
feat: port over wallet module

### PR DESCRIPTION
**Pull request by yunchipang**
## Description

**Issue(s) addressed**

- Resolves #

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**

**What is the new behavior?**

## Breaking change

**Does this PR introduce a breaking change?**

### CODE REVIEW

**Comment by yunchipang** [comment]
Screenshot 2024-09-13 at 9 36 18 AM

File: src/common/interfaces/limit-options.interface.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface LimitOptions {
```
Comment (Kpoke) [inline src/common/interfaces/limit-options.interface.ts:1]: Let's rename this to pagination options

File: src/common/services/s3.service.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+} from '@aws-sdk/client-s3';
+
+@Injectable()
+export class S3Service {
+  private readonly s3Client: S3Client;
+  private readonly bucket: string;
+  private readonly region: string;
+
+  constructor() {
+    this.bucket = process.env.S3_BUCKET || 'bucket'; // TODO: confirm this
```
Comment (Kpoke) [inline src/common/services/s3.service.ts:15]: Leave those as env variables with no defaults

**Comment by yunchipang** [comment]
thanks @Kpoke. will submit another PR for the changes

**Pull request merged**